### PR TITLE
[IMP] web: popover within iframes

### PR DESCRIPTION
**Pull request by brboi**

### CODE REVIEW

File: addons/web/static/src/core/utils/ui.js [hunk 1]
```diff
@@ -25,6 +25,20 @@ export function closest(elements, targetPos) {
...
+ * Check if an element is an HTMLElement, regardless of the window it belongs to.
+ * This is useful when we want to check if an element is an HTMLElement, but we
+ * don't know if it belongs to the current window or not.
+ * This is the case for example when the element to check is inside an iframe.
+ *
+ * @param {*} el
+ * @returns true if el is an HTMLElement, regardless of the window it belongs to.
+ */
+export function isHTMLElement(el) {
+    return HTML_ELEMENT_RE.test(el?.constructor.name);
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/ui.js:39]: Why using optional chaining here?

**Review by aab-odoo — APPROVED** [review]
Already looking good. When you think it's ready, maybe ask Leonardo to apply the commit in his branch to check if it works for him

File: addons/web/static/tests/core/popover/popover_tests.js [hunk 2]
```diff
@@ -158,3 +158,37 @@ QUnit.test("reposition popover should properly change classNames", async (assert
...
+    assert.containsOnce(fixture, ".o_popover");
+    assert.containsNone(iframe.contentDocument.documentElement, ".o_popover");
+
+    // The popover should be rendered in the correct position
+    const { top, left, width } = popoverEl.getBoundingClientRect();
+    const { top: targetTop, left: targetLeft } = popoverTarget.getBoundingClientRect();
+    const { top: iframeTop, left: iframeLeft } = iframe.getBoundingClientRect();
+    assert.strictEqual(top, iframeTop + targetTop + popoverTarget.offsetHeight);
+    assert.strictEqual(left, iframeLeft + targetLeft + popoverTarget.offsetWidth / 2 - width / 2);
+});
```
Comment (brboi) [inline addons/web/static/tests/core/popover/popover_tests.js:194]: should test iframe scrolling

**Comment by brboi** [comment]
closed in favor of: https://github.com/odoo/odoo/pull/116499

**Pull request closed without merging**